### ISSUE TITLE
Gateway Group support for static routes

### ISSUE THREAD

**Issue by JacBra** [opnsense/core#7626]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [x] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Is your feature request related to a problem? Please describe.**

For the purpose of prioritizing traffic over multiple IPsec tunnel (VTI) I am missing a solution to be able to statically route traffic over multiple tunnels". Currenty, only one static route towards a prefix can be defined.

**Describe the solution you like**

When Adding/Modifying a static route (System > Routes > Configuration), I would like to be able to set the input filed 'gateway' to a Gateway Group.

An alternative solution would be:
When Adding/Modifying a static route (System > Routes > Configuration), I would like to be able to add several routes for the same prefix, each with a different gateway. When the gateway is down, the route is disabled. When te gateway is up again, the route is enabled.

**Describe alternatives you considered**

As a workaround, I created a gateway group consisting of my IPsec gateways. Then, I used policy based routing in the firewall rules to achieve my goal. However, this is not so efficient, not flexible enough, and error-prone.

I also tried to use OSPF to dynamically route the traffic, but that did not work out in our use case: the firewall brand we use at the other side has serveral OSPF issues that led to hickups when doing configuration changes.

As a last resort, I ended up by routing a /24 subnet to the gateway of IPsecTunnel_1 and routing two /25 subnets to the gateway op IPsecTunnel_2. That works, but is very inflexible.

**Additional context**

![afbeelding](https://github.com/user-attachments/assets/9accdb28-85ff-4519-a1b9-7d1b592cfc7c)

**Comment by AdSchellevis** [comment]
gateway groups are intended for policy based routing only, the same type of information (weight/round-robin) and priority wouldn't apply to the routing table. These type of infrastructures usually require BGP or OSPF to handle multiple possible routing paths.

**Comment by JacBra** [comment]
I see, what about the alternative solution I formulated?

**Comment by AdSchellevis** [comment]
Some sort of mechanism to have a gateway monitor on them sounds nice, but is an awful lot of work which is likely not worth the effort (investment).

**Comment by JacBra** [comment]
I ended up by using a script that checks the gateways of every gateway group every 1 second. The script changes my static routes to the prefered gateway (based on highest Tier) that is 'up'.

**Comment by zubrick** [comment]
Yes, BGP or OSPF is a method of doing this, but all the major brands I came accros as a network engineer permit having priorities on routes and disable/enable them based on IP monitoring or interface status but technically it is mostly done in the software that changes the routes in the data plane based on that.
Mose useful being fortinet and juniper way of disabling all routes on interfaces that are not up, like a vpn virtual interface.

@JacBra would you mind sharing your script? it would be very helpful.

**Comment by JacBra** [comment]
@zubrick 
Sure. I use this only for gateway groups that contain my (multiple) IPsec interfaces. So not for WAN gateway groups

The solution I came up with is a script (`/usr/local/etc/rc.syshook.d/start/50-static_routes`) that runs at firewall boot. This script starts a second script (`/usr/local/bin/run_set_static_routes_gateway`), which contains an endless loop that calls a third script (`/usr/local/bin/set_static_routes_gateway`) every second.

You can change the preference by changing the 'Tier' values in a gateway group. And, when an IPsec tunnel goed down, it automagically switches fast to another route.

Note that it is essential that the name in the description of the static routes exactly matches the name of the Gateway Group.

I uploaded the files. Please remove the .rtf extensions for yourself.

[50-static-routes.rtf](https://github.com/user-attachments/files/29022875/50-static-routes.rtf)
[run_set_static_routes_gateway.rtf](https://github.com/user-attachments/files/29022874/run_set_static_routes_gateway.rtf)
[set_static_routes_gateway.rtf](https://github.com/user-attachments/files/29022873/set_static_routes_gateway.rtf)

**Comment by zubrick** [comment]
Thank you @JacBra
Exactly what I missed to use opnsense as a credible alternative to commercial brands

I modified slightly the set_static_routes function to allow still having a route description in the form of "gateway-group|route descrption" and to log using the log_message function instead of writing to standard output.

attached is my  version:
[set_static_routes_gateway.rtf](https://github.com/user-attachments/files/29058386/set_static_routes_gateway.rtf)
